internal/processors/cloudeventconvert/cloudeventconvert.go

```diff
@@ -265,22 +269,68 @@ func (c *cloudeventProcessor) createEventMsgs(origMsg *service.Message, source s
 	return messages, nil
 }
 
-func setDefaults(event *cloudevent.CloudEventHeader, source, defaultID string) {
+func validHeaderStrings(eventHdr *cloudevent.CloudEventHeader) error {
+
+	if !validCharacters.MatchString(eventHdr.ID) {
+		return errors.New("invalid header ID")
+	}
+
+	validSpec := regexp.MustCompile(`^[0-9.]+$`)
+	if !validSpec.MatchString(eventHdr.SpecVersion) {
+		return errors.New("invalid spec version")
+	}
+
+	validContentType := regexp.MustCompile(`^[a-zA-Z0-9\-_/]+$`)
+	if !validContentType.MatchString(eventHdr.DataContentType) {
+		return errors.New("invalid data content type")
+	}
+
+	return nil
+}
+
+func validateHeadersAndSetDefaults(event *cloudevent.CloudEventHeader, source, defaultID string) error {
 	event.Source = source
+
 	if event.Time.IsZero() {
 		event.Time = time.Now().UTC()
 	}
+
 	if event.ID == "" {
 		event.ID = defaultID
 	}
-
 	if event.SpecVersion == "" {
 		event.SpecVersion = "1.0"
 	}
-
 	if event.DataContentType == "" {
 		event.DataContentType = "application/json"
 	}
+
+	if !validCharacters.MatchString(event.ID) {
+		return errors.New("invalid header ID")
+	}
+	if !validCharacters.MatchString(event.SpecVersion) {
+		return errors.New("invalid spec version")
+	}
```

Comment:
nit: for these errors it might be nice to use the JSON name from the struct tag of the field we are checking
```suggestion
	if !validCharacters.MatchString(event.ID) {
		return errors.New("invalid id")
	}
	if !validCharacters.MatchString(event.SpecVersion) {
		return errors.New("invalid specversion")
	}
```